airflow/www/views.py

```diff
@@ -3574,13 +3600,32 @@ def datasets_summary(self):
                     DatasetModel.id,
                     DatasetModel.uri,
                 )
-                .filter(DatasetModel.uri.ilike(f"%{uri_pattern}%"))
                 .order_by(*order_by)
-                .offset(offset)
-                .limit(limit)
-                .all()
-            ]
-            data = {"datasets": datasets, "total_entries": total_entries}
+            )
+
+            if uri_pattern:
+                count_query = count_query.filter(DatasetModel.uri.ilike(f"%{uri_pattern}%"))
+                query = query.filter(DatasetModel.uri.ilike(f"%{uri_pattern}%"))
+
+            if updated_after:
+                count_query = count_query.outerjoin(
+                    DatasetEvent, DatasetEvent.dataset_id == DatasetModel.id
+                ).filter(DatasetEvent.timestamp >= updated_after)
+                query = query.filter(DatasetEvent.timestamp >= updated_after)
+            if updated_before:
+                count_query = count_query.outerjoin(
+                    DatasetEvent, DatasetEvent.dataset_id == DatasetModel.id
+                ).filter(DatasetEvent.timestamp <= updated_before)
+                query = query.filter(DatasetEvent.timestamp <= updated_before)
```

Comment:
if user provides both `before` and `after` filters, it looks like you would add two joins to the same table but you only need one.  you could do something like this instead.

```suggestion
            if uri_pattern:
                count_query = count_query.filter(DatasetModel.uri.ilike(f"%{uri_pattern}%"))
                query = query.filter(DatasetModel.uri.ilike(f"%{uri_pattern}%"))

            if updated_after:
                count_query = count_query.outerjoin(
                    DatasetEvent, DatasetEvent.dataset_id == DatasetModel.id
                ).filter(DatasetEvent.timestamp >= updated_after)
                query = query.filter(DatasetEvent.timestamp >= updated_after)
            if updated_before:
                count_query = count_query.outerjoin(
                    DatasetEvent, DatasetEvent.dataset_id == DatasetModel.id
                ).filter(DatasetEvent.timestamp <= updated_before)
                query = query.filter(DatasetEvent.timestamp <= updated_before)
```

Reply:
Or this I think would also work and is a bit cleaner:

```suggestion
            if updated_before or updated_after:
                count_query = count_query.outerjoin(DatasetEvent, DatasetEvent.dataset_id == DatasetModel.id)

            filters = []

            if uri_pattern:
                filters.append(DatasetModel.uri.ilike(f"%{uri_pattern}%"))

            if updated_after:
                filters.append(DatasetEvent.timestamp >= updated_after)

            if updated_before:
                filters.append(DatasetEvent.timestamp <= updated_before)

```

Reply:
I like the second option - seems cleaner to me. Thanks!

Reply:
@bbovenzi in the last commit you pushed, it seems that now you don't apply the filters to both queries